### DISCUSSION TITLE
`git record` doesn't move branch pointer

### DISCUSSION THREAD

**Discussion by talios** [arxanas/git-branchless#1242] [General]
For some reason or another, whenever I run `git record` I always end up on a detached HEAD commit, the parent commit is correct, and the stack seems fine, only, the branch reference never moves (hence being stuck on a detached HEAD).

Is this expected?

```
$ git sl
◇ 6546e6e 14d (develop, master) Initial commit
┃
◯ 4035a32 1h xxxxxxxxxxxxxxx
┃
◯ cce4f45 1h xxxxxxxxxxxxxxx
┃
◯ 3f4c8be 1h (feature/initial) xxxxxxxxxxxxxxx
┃
◯ 5c55f7a 1h xxxxxxxxxxxxxxx
┃
◯ 44a2198 1h xxxxxxxxxxxxxxx
┃
◯ ec73b10 1h xxxxxxxxxxxxxxx
┃
● 4014623 1h xxxxxxxxxxxxxxx
```

I could reset the branch ref by `git reset --hard 4014623` - but that feels.... wrong.

**Comment by claytonrcarter** [comment]
This behavior is intentional if you're already in a detached state (or you pass the `-d/--detach` option to `record`) - it only moves the branch pointer if you have the branch checked out when you run `record`. Can you confirm that you did, in fact, have `feature/initial` checked out?

**Reply by talios** [reply]
Mmm, I'm not sure off hand now - as it's been a few days, it's possible I'd made an additional commit during restacking/rebasing.

For now, I'll reset the branch pointer and keep an eye on it - now that I know it's supposed do unless already detached (which makes sense, and is kinda what I assumed).
